### PR TITLE
feat(deploy): Made changes in docker-compose, dockerfile and server to incorporate deployment requirements

### PR DESCRIPTION
**Pull request by oindrila-b**
- Modifed Dockerfile to build client side 
- Added a Dockerfile to deploy server/vespa image
- Modified server.ts to remove post-oauth redirect value
- Added a separate deploy-pod.sh to deploy vespa on a kubernetes pod.

### CODE REVIEW

File: deployment/docker-compose.dev.yml [hunk 1]
```diff
@@ -39,9 +39,9 @@ services:
     image: postgres
     container_name: xyne-db
     environment:
-      POSTGRES_USER: xyne
-      POSTGRES_PASSWORD: xyne
-      POSTGRES_DB: xyne
+      POSTGRES_USER: ${POSTGRES_USER}
```
Comment (zereraz) [inline deployment/docker-compose.dev.yml]: add xyne as default?

File: Dockerfile [hunk 2]
```diff
@@ -1,20 +1,19 @@
 # Use the official Bun image
 FROM oven/bun:1 AS base
+# Set working directory
```
Comment (zereraz) [inline Dockerfile]: remove this comment

File: Dockerfile [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
 WORKDIR /usr/src/app
 
-# Install backend dependencies
+# Copy your Bun app files
```
Comment (zereraz) [inline Dockerfile]: remove this comment

File: Dockerfile [hunk 4]
```diff
@@ -1,20 +1,19 @@
...
+COPY . .
+
+# Install dependencies for Bun server
```
Comment (zereraz) [inline Dockerfile]: remove this comment

File: Dockerfile [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
 WORKDIR /usr/src/app/server
-COPY server/package.json server/bun.lockb ./
-RUN bun install --frozen-lockfile
+RUN bun install
 
-# Install frontend dependencies
+# Install dependencies for Bun client
```
Comment (zereraz) [inline Dockerfile]: change it,

Install dependencies for `frontend`

File: deployment/docker-compose.dev.yml (hunk 1 above)
Comment (oindrila-b) [reply deployment/docker-compose.dev.yml]: Should this be just for the docker-compose.dev.yml or for all the docker composes?

Comment (zereraz) [reply deployment/docker-compose.dev.yml]: wherever we add variables

**Pull request merged**